Foundation/src/NestedDiagnosticContext.cpp

```diff
@@ -113,10 +113,14 @@ void NestedDiagnosticContext::clear()
 }
 
 
+namespace {
+	static ThreadLocal<NestedDiagnosticContext> ndc;
```

Comment:
See the comment of this class:
```cpp
class Foundation_API NestedDiagnosticContext
	/// This class implements a Nested Diagnostic Context (NDC),
	/// ......
	/// Every thread has its own private NDC, which is automatically
	/// created when needed and destroyed when the thread terminates.
	/// ......
{ ......
```
**"Every thread has its own private NDC".**

See the usage of this class:
```cpp
void thread1() {
    Poco::NDC::current().push("hello");
    Poco::NDC::current().push("world");
    Poco::NDC::current().toString(); // hello:world
}

void thread2() {
    Poco::NDC::current().push("hi");
    Poco::NDC::current().push("aleks-f");
    Poco::NDC::current().toString(); // hi:aleks-f
}
```
The method `NestedDiagnosticContext::current()` **must return a threadlocal variable for each thread**.

If it returns a singleton object:
1. It can't distinguish messages between threads. messages from all threads are mixed together.
2. It must add lock for thread safe.

Reply:
you are right; I got sidetracked by the previous implementation which had singleton, but that was just wrong way of doing it to start with